src/emqx_mod_topic_metrics.erl

```diff
@@ -180,7 +185,17 @@ val(Topic, Metric) ->
     end.
 
 rate(Topic, Metric) ->
-    gen_server:call(?MODULE, {get_rate, Topic, Metric}).
+    Rate = rates(Topic, Metric),
+    case is_tuple(Rate) of
```

Comment:
is_tuple isn't quite readable.
if we change the handle_call clause to return either `{ok, Rates}` or `{error, Reason}`,
it'd be easer to read here.